src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_targetid.lua

```diff
@@ -106,9 +106,14 @@ function GM:DrawSigilTargetHint(ent, fade)
 	util.ColorCopy(color_white, colTemp)
 
 	draw.SimpleTextBlur("Sigil", "ZSHUDFontSmaller", x, y, colTemp, TEXT_ALIGN_CENTER)
-	y = y + draw.GetFontHeight("ZSHUDFontSmaller") + 0
+	local fontHeight = draw.GetFontHeight("ZSHUDFontSmaller")
+	y = y + fontHeight + 0
 
 	draw.SimpleTextBlur("Press E to teleport", "ZSHUDFontTiny", x, y, colTemp, TEXT_ALIGN_CENTER)
+
+	if MySelf:IsValid() and MySelf:Team() == TEAM_HUMAN and self:PlayerCanPurchase(MySelf) then
```

Comment:
The first conditions are unnecessary since `GM:DrawSigilTargetHint` is only invoked for humans:
 https://github.com/rbreslow/zs/blob/cb6206b43c8ea4896305cf26979de75a837fce23/src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_targetid.lua#L149-L151

Additionally, I don't think it's necessary to check `MySelf:IsValid()` because order rendering hooks seem to think it's fine. e.g.

https://github.com/rbreslow/zs/blob/00f1a80f382d72ceb91912941587e67b5b735e05/src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_init.lua#L853-L858

`MySelf` is defined in `GM:InitPostEntity` so there won't be many cases where it's invalid.

Reply:
You're right; these checks have been removed.